modules/search/class.jetpack-search-helpers.php

```diff
@@ -680,4 +680,161 @@ public static function get_max_posts_per_page() {
 	public static function get_max_offset() {
 		return Jetpack_Search_Options::site_has_vip_index() ? 9000 : 1000;
 	}
+
+	/**
+	 * Returns the maximum offset for a search query.
+	 *
+	 * @since 8.4.0
+	 * @param string $locale    A potentially valid locale string.
+	 *
+	 * @return bool
+	 */
+	public static function is_valid_locale( $locale ) {
+		// Sourced from https://wpastra.com/docs/complete-list-wordpress-locale-codes/.
+		$valid_locales = array(
```

Comment:
Is there a better place to put this list than inside this helper class?

Reply:
I think this location does make sense. Can we use https://github.com/Automattic/jetpack/blob/7923f4ac1c6f2974eca870f8b64238b2fa204d8c/locales.php instead of hard coding our own?

This does seem like a good solution.

Reply:
Updated to use GlotPress for locale validation in 
b19bfda.